lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -123,15 +139,28 @@ class Scripts extends FRGatherer {
       // It's nice to display the user-provided value in Lighthouse, so we add a field 'name'
       // to make it clear this is for presentational purposes.
       // See https://chromium-review.googlesource.com/c/v8/v8/+/2317310
+      let name = event.url;
+      // embedderName is optional on the protocol because backends like Node may not set it.
+      // For our purposes, it is always set. But just in case it isn't... fallback to the url.
+      let url = event.embedderName || event.url;
+
+      // Some eval'd scripts may not have a name or url, so let's set them.
+      if (!name) name = '<eval script>';
+      if (!url && event.stackTrace?.callFrames.length) url = event.stackTrace.callFrames[0].url;
+      if (!url) url = this._scriptFrameUrls[i] || '';
```

Comment:
If it's just for debugging, could we make it part of `name` instead of `url`? I feel like this gives `url` two different meanings depending on if the script was compiled from a string or not.

Reply:
Good point. This also would have added these dynamic scripts to byte efficiency audits, which likely would have been totally nonsensical to think whatever reduction in the script would correlate with a byte reduction in the frame url that added it.

so let's make `Script.url` optional and force audits to deal with it.